### ISSUE TITLE
Problem with SAEM covariate modeling

### ISSUE THREAD

**Issue by roninsightrx** [nlmixr2/nlmixr2est#348]
hi Matt,

I'm having an issue when running SAEM in nlmixr2, hope you can help me.

I'm run a simple 2-cmt PK model and testing various covariates. The issue is that SAEM is not giving me sensible results,
basically ignoring any covariate I try to test and just sampling a dummy parameter (it seems)... 

Let's say I have the two models below, model 5 testing CRCL as a covariate, and model 6 testing WT. I'm using linear solved model here, but the same issue occurs with ODE systems.

```
mod5 <- function() {
  ini({
    tCL     <- log(5)
    tV      <- log(50)
    tQ      <- log(5)
    tV2     <- log(100)
    tCRCL_CL <- 0.5
    eta_CL   ~ 1.0
    eta_V    ~ 1.0
    add_sd <- 0.5
  })
  model({
    CL <- exp(tCL + eta_CL + logCRCL100*tCRCL_CL )
    V  <- exp(tV  + eta_V)
    Q  <- exp(tQ)
    V2 <- exp(tV2)
    linCmt() ~ add(add_sd)
  })
}

mod6 <- function() {
  ini({
    tCL     <- log(5)
    tV      <- log(50)
    tQ      <- log(5)
    tV2     <- log(100)
    tWT_CL  <- 0.5
    eta_CL  ~ 1.0
    eta_V   ~ 1.0
    add_sd <- 0.5
  })
  model({
    CL <- exp(tCL + eta_CL + logWT70*tWT_CL)
    V  <- exp(tV  + eta_V)
    Q  <- exp(tQ)
    V2 <- exp(tV2)
    linCmt() ~ add(add_sd)
  })
}
```
Note the different covariate estimate.
The head of my dataset looks like below. Obviously logCRCL100 and logWT70 are not the same. I can send you the full data via email if you like.
```
  ID    TIME   DV EVID MDV CMT  AMT RATE   logWT70  logFFM70 logCRCL100 caplogCRCL100       logCR   logCRstd
1  1  0.0000  0.0    1   1   1 2500 1000 0.6411017 0.1107744  0.6042130     0.4054651  0.00000000 -0.1457015
2  1  3.7000 27.9    0   0   1    0    0 0.6411017 0.1107744  0.6042130     0.4054651  0.00000000 -0.1457015
3  1 10.9167  8.6    0   0   1    0    0 0.6411017 0.1107744  0.6042130     0.4054651  0.00000000 -0.1457015
4  1 11.4500  0.0    1   1   1 1500 1000 0.6411017 0.1107744  0.6042130     0.4054651  0.00000000 -0.1457015
5  1 19.6000  0.0    1   1   1 1500 1000 0.6411017 0.1107744  0.6875946     0.4054651 -0.08338161 -0.1457015
6  1 27.2500  0.0    1   1   1 1500 1000 0.6411017 0.1107744  0.6875946     0.4054651 -0.08338161 -0.1457015
```

When I then sample:
```
res5 <- nlmixr(
  mod5, 
  data, 
  est = "saem", control = list(nBurn=50, nEm=50, print=25)
)
res6 <- nlmixr(
  mod6, 
  data, 
  est = "saem", control = list(nBurn=50, nEm=50, print=25)
)
```

The parameter estimates are exactly the same 
```
> res5$parFixed
          Est.     SE %RSE Back-transformed(95%CI) BSV(CV%) Shrink(SD)%
tCL       1.43  0.186 13.1       4.16 (2.89, 5.99)     72.3      3.02%<
tV        2.02 0.0382 1.89       7.57 (7.02, 8.16)     34.1      16.9%<
tQ        4.08 0.0686 1.68       59.2 (51.8, 67.7)                     
tV2      0.577  0.065 11.3       1.78 (1.57, 2.02)                     
tCRCL_CL  4.06 0.0429 1.06       4.06 (3.98, 4.14)                     
add_sd    2.83                                2.83                     
> res6$parFixed
        Est.     SE %RSE Back-transformed(95%CI) BSV(CV%) Shrink(SD)%
tCL     1.43  0.186 13.1       4.16 (2.89, 5.99)     72.3      3.02%<
tV      2.02 0.0382 1.89       7.57 (7.02, 8.16)     34.1      16.9%<
tQ      4.08 0.0686 1.68       59.2 (51.8, 67.7)                     
tV2    0.577  0.065 11.3       1.78 (1.57, 2.02)                     
tWT_CL  4.06 0.0429 1.06       4.06 (3.98, 4.14)                     
add_sd  2.83                                2.83          
```
One interesting thing, and perhaps related, is that the estimates are actually wrongly reported in the table here. CL is correct, but the order of the other estimates are all wrong.

Do you have any idea what is going on here?

**Comment by mattfidler** [comment]
The warning now is shown below

``` r
devtools::load_all("~/src/nlmixr2est")
#> ℹ Loading nlmixr2est
#> Loading required package: nlmixr2data
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> 
#> The following object is masked from 'package:testthat':
#> 
#>     matches
#> 
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> 
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(ggplot2)
#library(nlmixr2)

## Data prep
# data <- read.csv(file = "./data/nmdata_20230216_1.csv") %>%
#   mutate(CRCLcap = ifelse(CRCL < 150, CRCL, 150)) %>%
#   mutate(logCR = log(CREAT/1.0)) %>%
#   mutate(logWT70 = log(WEIGHT/70)) %>%
#   mutate(logCRCL100 = log(CRCL/100)) %>%
#   mutate(caplogCRCL100 = log(CRCLcap/100)) %>%
#   mutate(logCRCLf100 = log(CRCLf/100)) %>%
#   mutate(logCRstd = -1.22839 + log10(AGE) * 0.67190 + 6.27017 * exp(-3.10940 * AGE)) %>%
#   filter(EVID != 2) %>%
#   mutate(CMT = 1) %>%
#   select(ID, TIME, DV, EVID, MDV, CMT, AMT, RATE, logWT70, logCRCL100)
# write.csv(data, file = "./data/data_matt.csv", quote=F, row.names=F)

data <- read.csv(file="~/Downloads/data_matt.csv")


mod5 <- function() {
  ini({
    tCL     <- log(5)
    tV      <- log(50)
    tQ      <- log(5)
    tV2     <- log(100)
    tCRCL_CL <- 0.5
    eta_CL   ~ 1.0
    eta_V    ~ 1.0
    add_sd <- 0.5
  })
  model({
    CL <- exp(tCL + eta_CL + logCRCL100*tCRCL_CL)
    V  <- exp(tV  + eta_V)
    Q  <- exp(tQ)
    V2 <- exp(tV2)
    linCmt() ~ add(add_sd)
  })
}

mod6 <- function() {
  ini({
    tCL     <- log(5)
    tV      <- log(50)
    tQ      <- log(5)
    tV2     <- log(100)
    tWT_CL  <- 0.5
    eta_CL  ~ 1.0
    eta_V   ~ 1.0
    add_sd <- 0.5
  })
  model({
    CL <- exp(tCL + eta_CL + logWT70*tWT_CL)
    V  <- exp(tV  + eta_V)
    Q  <- exp(tQ)
    V2 <- exp(tV2)
    linCmt() ~ add(add_sd)
  })
}


res5 <- nlmixr(
  mod5, 
  data, 
  est = "saem", control = saemControl(nBurn=50, nEm=50, print=25)
)
#> → loading into symengine environment...
#> → pruning branches (`if`/`else`) of saem model...
#> ✔ done
#> → finding duplicate expressions in saem model...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> ✔ done
#> params:  tCL tV  tQ  tV2 tCRCL_CL    V(eta_CL)   V(eta_V)    add_sd
#> rxode2 2.0.13.9000 using 8 threads (see ?getRxThreads)
#>   no cache: create with `rxCreateCache()`
#> 001: 1.540419    4.012219    1.614770    4.494998    0.420887    0.950000    0.950000    6.415708    
#> 025: 1.386858    4.137677    1.763550    4.210289    0.626610    0.444569    0.277390    2.728454    
#> 050: 1.418467    4.059830    1.998436    3.999648    0.606273    0.381839    0.105896    2.815682    
#> 075: 1.422673    4.067952    2.025585    4.077568    0.570257    0.400934    0.102184    2.818364    
#> 100: 1.425228    4.059580    2.023862    4.081245    0.576987    0.420640    0.109775    2.825762    
#> Calculating covariance matrix
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00 
#> Error : Error calculating covariance via linearization
#> → loading into symengine environment...
#> → pruning branches (`if`/`else`) of saem model...
#> ✔ done
#> → finding duplicate expressions in saem predOnly model 0...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → finding duplicate expressions in saem predOnly model 1...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → finding duplicate expressions in saem predOnly model 2...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> ✔ done
#> → Calculating residuals/tables
#> ✔ done
#> → compress origData in nlmixr2 object, save 45216
#> → compress phiM in nlmixr2 object, save 135560
#> → compress parHist in nlmixr2 object, save 3408


res5$parFixed
#>           Est.     SE %RSE Back-transformed(95%CI) BSV(CV%) Shrink(SD)%
#> tCL       1.43  0.186 13.1       4.16 (2.89, 5.99)     72.3      3.02%<
#> tV        2.02 0.0382 1.89       7.57 (7.02, 8.16)     34.1      16.9%<
#> tQ        4.08 0.0686 1.68       59.2 (51.8, 67.7)                     
#> tV2      0.577  0.065 11.3       1.78 (1.57, 2.02)                     
#> tCRCL_CL  4.06 0.0429 1.06       4.06 (3.98, 4.14)                     
#> add_sd    2.83                                2.83

res5$runInfo
#> [1] "IDs without observations dropped: 97 89 86 85 84 83 76 67 62 59 58 35 30 27 23 3"                                                                                                                 
#> [2] "covariance matrix non-positive definite, corrected by sqrtm(fim %*% fim)"                                                                                                                         
#> [3] "NAs introduced by coercion"                                                                                                                                                                       
#> [4] "log-scale mu referenced time varying covariates (tCRCL_CL) may have better results on no log-transformed scale (https://github.com/nlmixr2/nlmixr2est/issues/348), check results for plausibility"
```

<sup>Created on 2023-05-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">

<summary>Session info</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.0 (2023-04-21)
#>  os       Pop!_OS 22.04 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/Chicago
#>  date     2023-05-06
#>  pandoc   2.9.2.1 @ /usr/bin/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  ! package       * version     date (UTC) lib source
#>    backports       1.4.1       2021-12-13 [3] RSPM (R 4.2.0)
#>    brio            1.1.3       2021-11-30 [3] RSPM (R 4.2.0)
#>    cachem          1.0.8       2023-05-01 [3] RSPM (R 4.2.0)
#>    callr           3.7.3       2022-11-02 [3] RSPM (R 4.2.0)
#>    checkmate       2.2.0       2023-04-27 [3] RSPM (R 4.2.0)
#>    cli             3.6.1       2023-03-23 [3] RSPM (R 4.2.0)
#>    colorspace      2.1-0       2023-01-23 [3] RSPM (R 4.2.0)
#>    crayon          1.5.2       2022-09-29 [3] RSPM (R 4.2.0)
#>    data.table      1.14.8      2023-02-17 [3] RSPM (R 4.2.0)
#>    desc            1.4.2       2022-09-08 [3] RSPM (R 4.2.0)
#>    devtools        2.4.5       2022-10-11 [3] RSPM (R 4.2.0)
#>    digest          0.6.31      2022-12-11 [3] RSPM (R 4.2.0)
#>    dparser         1.3.1-10    2023-03-16 [3] RSPM (R 4.2.0)
#>    dplyr         * 1.1.2       2023-04-20 [3] RSPM (R 4.2.0)
#>    ellipsis        0.3.2       2021-04-29 [3] RSPM (R 4.2.0)
#>    evaluate        0.20        2023-01-17 [3] RSPM (R 4.2.0)
#>    fansi           1.0.4       2023-01-22 [3] RSPM (R 4.2.0)
#>    fastmap         1.1.1       2023-02-24 [3] RSPM (R 4.2.0)
#>    fs              1.6.2       2023-04-25 [3] RSPM (R 4.2.0)
#>    generics        0.1.3       2022-07-05 [3] RSPM (R 4.2.0)
#>    ggplot2       * 3.4.2       2023-04-03 [3] RSPM (R 4.2.0)
#>    glue            1.6.2       2022-02-24 [3] RSPM (R 4.2.0)
#>    gtable          0.3.3       2023-03-21 [3] RSPM (R 4.2.0)
#>    htmltools       0.5.5       2023-03-23 [3] RSPM (R 4.2.0)
#>    htmlwidgets     1.6.2       2023-03-17 [3] RSPM (R 4.2.0)
#>    httpuv          1.6.9       2023-02-14 [3] RSPM (R 4.2.0)
#>    knitr           1.42        2023-01-25 [3] RSPM (R 4.2.0)
#>    later           1.3.1       2023-05-02 [3] RSPM (R 4.2.0)
#>    lattice         0.21-8      2023-04-05 [3] RSPM (R 4.2.0)
#>    lazyeval        0.2.2       2019-03-15 [3] RSPM (R 4.2.0)
#>    lbfgsb3c        2020-3.2    2020-03-03 [3] RSPM (R 4.2.0)
#>    lifecycle       1.0.3       2022-10-07 [3] RSPM (R 4.2.0)
#>    lotri           0.4.3       2023-03-20 [3] RSPM (R 4.2.0)
#>    magrittr        2.0.3       2022-03-30 [3] RSPM (R 4.2.0)
#>    Matrix          1.5-4       2023-04-04 [3] RSPM (R 4.2.0)
#>    memoise         2.0.1       2021-11-26 [3] RSPM (R 4.2.0)
#>    mime            0.12        2021-09-28 [3] RSPM (R 4.2.0)
#>    miniUI          0.1.1.1     2018-05-18 [3] RSPM (R 4.2.0)
#>    minqa           1.2.5       2022-10-19 [3] RSPM (R 4.2.0)
#>    munsell         0.5.0       2018-06-12 [3] RSPM (R 4.2.0)
#>    n1qn1           6.0.1-11    2022-10-18 [3] RSPM (R 4.2.0)
#>    nlme            3.1-162     2023-01-31 [3] RSPM (R 4.2.0)
#>    nlmixr2data   * 2.0.7       2022-04-22 [3] RSPM (R 4.2.0)
#>  P nlmixr2est    * 2.1.5       2023-05-06 [?] load_all()
#>    numDeriv        2016.8-1.1  2019-06-06 [3] RSPM (R 4.2.0)
#>    optextras       2019-12.4   2019-12-20 [3] RSPM (R 4.2.0)
#>    pillar          1.9.0       2023-03-22 [3] RSPM (R 4.2.0)
#>    pkgbuild        1.4.0       2022-11-27 [3] RSPM (R 4.2.0)
#>    pkgconfig       2.0.3       2019-09-22 [3] RSPM (R 4.2.0)
#>    pkgload         1.3.2       2022-11-16 [3] RSPM (R 4.2.0)
#>    PreciseSums     0.6         2023-04-22 [3] RSPM (R 4.2.0)
#>    prettyunits     1.1.1       2020-01-24 [3] RSPM (R 4.2.0)
#>    processx        3.8.1       2023-04-18 [3] RSPM (R 4.2.0)
#>    profvis         0.3.8       2023-05-02 [3] RSPM (R 4.2.0)
#>    promises        1.2.0.1     2021-02-11 [3] RSPM (R 4.2.0)
#>    ps              1.7.5       2023-04-18 [3] RSPM (R 4.2.0)
#>    purrr           1.0.1       2023-01-10 [3] RSPM (R 4.2.0)
#>    qs              0.25.5      2023-02-22 [3] RSPM (R 4.2.0)
#>    R.cache         0.16.0      2022-07-21 [3] RSPM (R 4.2.0)
#>    R.methodsS3     1.8.2       2022-06-13 [3] RSPM (R 4.2.0)
#>    R.oo            1.25.0      2022-06-12 [3] RSPM (R 4.2.0)
#>    R.utils         2.12.2      2022-11-11 [3] RSPM (R 4.2.0)
#>    R6              2.5.1       2021-08-19 [3] RSPM (R 4.2.0)
#>    RApiSerialize   0.1.2       2022-08-25 [3] RSPM (R 4.2.0)
#>    Rcpp            1.0.10      2023-01-22 [3] RSPM (R 4.2.0)
#>    RcppParallel    5.1.7       2023-02-27 [3] RSPM (R 4.2.0)
#>    remotes         2.4.2       2021-11-30 [3] RSPM (R 4.2.0)
#>    reprex          2.0.2       2022-08-17 [3] RSPM (R 4.2.0)
#>    rex             1.2.1       2021-11-26 [3] RSPM (R 4.2.0)
#>    rlang           1.1.1       2023-04-28 [3] RSPM (R 4.2.0)
#>    rmarkdown       2.21        2023-03-26 [3] RSPM (R 4.2.0)
#>    rprojroot       2.0.3       2022-04-02 [3] RSPM (R 4.2.0)
#>    rstudioapi      0.14        2022-08-22 [3] RSPM (R 4.2.0)
#>    Rvmmin          2018-4.17.1 2021-04-18 [3] RSPM (R 4.2.0)
#>    rxode2        * 2.0.13.9000 2023-05-04 [1] local
#>    rxode2et        2.0.10      2023-03-17 [3] RSPM (R 4.2.0)
#>    rxode2ll        2.0.11      2023-03-17 [3] RSPM (R 4.2.0)
#>    rxode2parse     2.0.16.9000 2023-05-04 [1] local
#>    rxode2random    2.0.11.9000 2023-05-04 [1] local
#>    scales          1.2.1       2022-08-20 [3] RSPM (R 4.2.0)
#>    sessioninfo     1.2.2       2021-12-06 [3] RSPM (R 4.2.0)
#>    shiny           1.7.4       2022-12-15 [3] RSPM (R 4.2.0)
#>    stringfish      0.15.7      2022-04-13 [3] RSPM (R 4.2.0)
#>    stringi         1.7.12      2023-01-11 [3] RSPM (R 4.2.0)
#>    stringr         1.5.0       2022-12-02 [3] RSPM (R 4.2.0)
#>    styler          1.9.1       2023-03-04 [3] RSPM (R 4.2.0)
#>    symengine       0.2.2       2022-10-23 [3] RSPM (R 4.2.0)
#>    sys             3.4.1       2022-10-18 [3] RSPM (R 4.2.0)
#>    testthat      * 3.1.8       2023-05-04 [3] RSPM (R 4.2.0)
#>    tibble          3.2.1       2023-03-20 [3] RSPM (R 4.2.0)
#>    tidyselect      1.2.0       2022-10-10 [3] RSPM (R 4.2.0)
#>    ucminf          1.1-4.1     2022-09-29 [3] RSPM (R 4.2.0)
#>    units           0.8-2       2023-04-27 [3] RSPM (R 4.2.0)
#>    urlchecker      1.0.1       2021-11-30 [3] RSPM (R 4.2.0)
#>    usethis         2.1.6       2022-05-25 [3] RSPM (R 4.2.0)
#>    utf8            1.2.3       2023-01-31 [3] RSPM (R 4.2.0)
#>    vctrs           0.6.2       2023-04-19 [3] RSPM (R 4.2.0)
#>    withr           2.5.0       2022-03-03 [3] RSPM (R 4.2.0)
#>    xfun            0.39        2023-04-20 [3] RSPM (R 4.2.0)
#>    xtable          1.8-4       2019-04-21 [3] RSPM (R 4.2.0)
#>    yaml            2.3.7       2023-01-23 [3] RSPM (R 4.2.0)
#> 
#>  [1] /home/matt/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /usr/local/lib/R/site-library
#>  [3] /usr/lib/R/site-library
#>  [4] /usr/lib/R/library
#> 
#>  P ── Loaded and on-disk path mismatch.
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by mattfidler** [comment]
This may have been fixed with b8c7292f9f3310bfcada5e331a008820291e0d6c
